### PR TITLE
Build

### PR DESCRIPTION
**Pull request by LeSpocky**
Some more or less trivial build fixes. Makes it a little easier for others to try this out and play with it.

### CODE REVIEW

File: mqtt-clock.c [hunk 1]
```diff
@@ -168,14 +168,13 @@ int main(int argc, char *argv[]) {
 
   if (agent) {
     int ret;
-    mosqagent_add_idle_call(agent, clock_idle);
+    ret = mosqagent_add_idle_call(agent, clock_idle);
```
Comment (LeSpocky) [inline mqtt-clock.c]: I think this is a real bugfix and should be in an extra commit.

File: mqtt-clock.c [hunk 2]
```diff
@@ -167,14 +168,13 @@ int main(int argc, char *argv[]) {
...
 
     if (!ret)
       syslog(LOG_ERR, "Cannot add clock idle call: %s",
 	     mosqagent_strerror(ret));
   }
 
-  int ret;
```
Comment (penguineer) [inline mqtt-clock.c]: This is pure clean-up, should also be an extra commit.

**Comment by penguineer** [comment]
Thanks for the PR. I have a not-yet-pushed branch of a further development state for this library, which I will check against your proposal. 

I'll comment when this is done (to avoid unnecessary work).

**Review by penguineer — CHANGES_REQUESTED** [review]
Please split this commit into:
1) One commit fixing the code
2) One commit changing the build environment.

Either in this PR or as a new PR. I'll integrate the result.
Thanks.

**Comment by LeSpocky** [comment]
Splitted the requested commit. Note: there are still compiler warnings left, I did not fix, all of the type _return-type_ aka "control reaches end of non-void function".

The whitespace changes in 6166b04bca14615feb81db83c6361641f94b4433 were done automatically by my editor. Hope that doesn't disturb too much.

**Review by penguineer — APPROVED** [review]

**Comment by penguineer** [comment]
Thanks, merged and closed.

(nb: Most editors can be set to fix whitespace issues only in lines which have been changed anyway.)

**Pull request merged**